js/freewheel.js

```diff
@@ -283,6 +315,8 @@ OO.Ads.manager(function(_, $) {
       } else {
         OO.log("FW: freewheel metadata request failure");
       }
+      // clear ad request timeout since fw_onAdRequestComplete was called
+      _clearAdRequestTimeout();
```

Comment:
Can you move this to the top of the function?  That way if someone ads code and it happens to exit early, we don't forget to do this.